tool/net/redbean.c

```diff
@@ -1915,6 +1915,26 @@ static int LuaGetZipPaths(lua_State *L) {
   return 1;
 }
 
+static int LuaSetHandler(lua_State *L) {
+  int t = lua_type(L, 1);
+
+  if(t == LUA_TFUNCTION) {
+    lua_setglobal(L, "Handler");
```

Comment:
This goes along with the hotpath comment, below. If we expose it as `Handler` then a user should be able to expect to change their handler by reassigning the global, or be able to introspect the handler function through Lua's debug tools. (Like accessing the upvalues).

Something like:

```lua

-- Change behaviour if there is a handler
if Handler ~= nil then
  Write("Handler not in use.")
end

-- Reassign handler
Handler = function()
  Write("New handler!")
end

-- Inspect handler
local i = 1
while true do
  -- see if i is a valid upvalue index
  local name = debug.getupvalue(Handler, i)
  if not name then
    break
  end
  print(name)

  i = i + 1
end
```

Reply:
Sure we can go that route. In that case, I'd suggest deleting `SetHandler` since less API surface area is better. We might also consider naming this `OnHttpRequest` since the API will likely evolve over time to introduce other kinds of event handlers, such as `OnHeartbeat` or `OnFork`.